src/sales.ts

```diff
@@ -0,0 +1,126 @@
+import { AuctionHouseProgram } from '@metaplex-foundation/mpl-auction-house'
+import {
+  createPrintListingReceiptInstruction,
+  createSellInstruction,
+} from '@metaplex-foundation/mpl-auction-house/dist/src/generated/instructions'
+import { Wallet } from '@metaplex/js'
+import {
+  Connection,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  SYSVAR_INSTRUCTIONS_PUBKEY,
+  Transaction,
+} from '@solana/web3.js'
+import { Client } from './client'
+import { AuctionHouse, Nft } from './types'
+
+export interface SellParams {
+  amount: number
+  nft: Nft
+}
+
+export class SalesClient extends Client {
+  private auctionHouse: AuctionHouse
+
+  constructor(
+    connection: Connection,
+    wallet: Wallet,
+    auctionHouse: AuctionHouse
+  ) {
+    super(connection, wallet)
+
+    this.auctionHouse = auctionHouse
+  }
+
+  async sell({ amount, nft }: SellParams) {
+    const { publicKey, signTransaction } = this.wallet
+    const connection = this.connection
+    const ah = this.auctionHouse
+    const buyerPrice = Number(amount) * LAMPORTS_PER_SOL
+    const auctionHouse = new PublicKey(ah.address)
+    const authority = new PublicKey(ah.authority)
```

Comment:
We already have listing create. This is redundant. Please drop the sales client.